buffer/src/main/java/io/netty/buffer/FixedCompositeByteBuf.java

```diff
@@ -290,6 +334,18 @@ protected int _getInt(int index) {
     }
 
     @Override
+    protected int _getIntLE(int index) {
+        Component c = findComponent(index);
+        if (index + 4 <= c.endOffset) {
+            return c.buf.getIntLE(index - c.offset);
+        } else if (order() == ByteOrder.BIG_ENDIAN) {
+            return _getShort(index) & 0xFFFF | (_getShort(index + 2) & 0xFFFF) << 16;
+        } else {
+            return (_getShort(index) & 0xffff) << 16 | _getShort(index + 2) & 0xffff;
+        }
+    }
```

Comment:
Problem with extracting these to the common util class is that protected methods `_getShort` and so on are used. Also, that `Component` class is private (there are two of them, one is readonly and second one is read/write).